### PR TITLE
Feature/Adding-additional-data-points-for-historical-appointments

### PR DESCRIPTION
**Pull request by nobert-mumo**
Adding additional columns to the linelist appointments for Hifadhi reporting.

### CODE REVIEW

File: Scripts/NDWH/C&T FACT TABLES/load_FactAppointments.sql [hunk 1]
```diff
@@ -1,24 +1,22 @@
 IF OBJECT_ID(N'[NDWH].[dbo].FACTAppointments', N'U') IS NOT NULL 		
 	drop table [NDWH].[dbo].FACTAppointments
 GO
-With Patients As (
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactAppointments.sql:4]: deleted this as it was not being used anywhere

File: Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+---declare your start and end dates.
 declare 
 @start_date date = '2020-06-01',
 @end_date date = '2023-06-30';
 
 with dates as (
-
-    
-      
+
```
Comment (Marymary-dev) [inline Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql:14]: @nobert-mumo Remember to include the Dec 2023 End date

**Review by Marymary-dev — APPROVED** [review]

File: Scripts/NDWH/C&T FACT TABLES/load_FactAppointments.sql [hunk 3]
```diff
@@ -33,17 +31,22 @@ Select
...
-		
-        into NDWH.dbo.FACTAppointments
-        from ODS.dbo.[HistoricalAppointmentStatus] as apt
+		RegimenAsof,
+		coalesce(NoOfUnscheduledVisits, 0) as NoOfUnscheduledVisitsAsOf,
+        patient_info.StartARTDate,
+        patient_info.Patienttype,
+        cast(getdate() as date) as LoadDate		
+into NDWH.dbo.FACTAppointments
+from tmp_and_adhoc.dbo.[HistoricalAppointmentStatus] as apt
```
Comment (nthusi-codes) [inline Scripts/NDWH/C&T FACT TABLES/load_FactAppointments.sql:40]: Is tmp_and_adhoc meant to be the final source db?

**Review by nthusi-codes — APPROVED** [review]
Looks good. Though not sure about the use of tmp_and_adhoc as it's not one of the official db stages. i.e (ODS, EDW, REPORTING)

**Comment by nobert-mumo** [comment]
> Looks good. Though not sure about the use of tmp_and_adhoc as it's not one of the official db stages. i.e (ODS, EDW, REPORTING)

Hi @nthusi-codes, nice observation indeed. So the idea was that we should be able to build the historical datasets on demand. I decided to build this on tmp_and_adhoc  because it is the only server we have read access in the .83 server. But I see your point of using tmp_and_adhoc whhich should ideally be a temp area. Maybe we can have a special DB to store this historical tables that we run on demand? WDYT ?

**Comment by Marymary-dev** [comment]
I agree with the proposal to create the table into a different schema that is not tmp_and_adhoc

**Comment by nobert-mumo** [comment]
> I agree with the proposal to create the table into a different schema that is not tmp_and_adhoc

Indeed I am going to move the historical table to the DB `Historical` & update this PR accordingly

**Comment by nthusi-codes** [comment]
> > Looks good. Though not sure about the use of tmp_and_adhoc as it's not one of the official db stages. i.e (ODS, EDW, REPORTING)
> 
> Hi @nthusi-codes, nice observation indeed. So the idea was that we should be able to build the historical datasets on demand. I decided to build this on tmp_and_adhoc because it is the only server we have read access in the .83 server. But I see your point of using tmp_and_adhoc whhich should ideally be a temp area. Maybe we can have a special DB to store this historical tables that we run on demand? WDYT ?

I like the historical db idea. I just didn't get why tmp was being used.

**Comment by nobert-mumo** [comment]
@nthusi-codes , @Marymary-dev, I made the update to replace building the history tables from `tmp_and_adhoc` db to `Historical` db. You can have a look.

**Review by cbrianbet — APPROVED** [review]

**Pull request merged**